gtdynamics/dynamics/ChainDynamicsGraph.cpp

```diff
@@ -0,0 +1,223 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file DynamicsGraph.h
+ * @brief Builds a lean dynamics graph from a Robot object.
+ * @author Dan Barladeanu
+ */
+
+#include "gtdynamics/dynamics/ChainDynamicsGraph.h"
+
+namespace gtdynamics {
+
+std::vector<std::vector<JointSharedPtr>> ChainDynamicsGraph::getChainJoints(const Robot& robot) {
+  std::vector<JointSharedPtr> FR(3), FL(3), RR(3), RL(3);
+
+  int loc = 0;
+  for (auto&& joint : robot.joints()) {
+    if (joint->name().find("lower") != std::string::npos) {
```

Comment:
This seems very specific to the A1. Which isn't great since `Robot` can be any legged robot like the Anymal C right?